tiledb/api/c_api/current_domain/current_domain_api_external_experimental.h

```diff
@@ -0,0 +1,185 @@
+ * Set a N-dimensional rectangle representation on a current domain.
+ * Error if the current domain passed is not empty.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_current_domain_t *current_domain;
+ * tiledb_current_domain_create(ctx, &current_domain);
+ * tiledb_ndrectangle_t *ndr;
+ * tiledb_ndrectangle_alloc(ctx, domain, &ndr);
+ *
+ * tiledb_range_t range;
+ * range.min = &min;
+ * range.min_size = sizeof(min);
+ * range.max = &max;
+ * range.max_size = sizeof(max);
+ * tiledb_ndrectangle_set_range_for_name(ctx, ndr, "dim", &range);
+ *
+ * tiledb_current_domain_set_ndrectangle(current_domain, ndr);
+ *
+ * tiledb_ndrectangle_free(&ndr);
+ * tiledb_current_domain_free(&current_domain);
+ * @endcode
+ *
+ * @param current_domain The current domain to modify
+ * @param ndr The N-dimensional rectangle to be set
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT capi_return_t tiledb_current_domain_set_ndrectangle(
+    tiledb_current_domain_t* current_domain,
+    tiledb_ndrectangle_t* ndr) TILEDB_NOEXCEPT;
+
+/**
+ * Get the N-dimensional rectangle associated with the current domain object.
+ * Error if the current domain is empty or a different representation is set.
+ *
+ * It is the responsability of the caller to free the resources associated
+ * with the ndrectangle when the handle isn't needed anymore.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_ndrectangle_t *ndr;
+ * tiledb_current_domain_get_ndrectangle(current_domain, &ndr);
+ * @endcode
+ *
+ * @param current_domain The current domain to query
+ * @param ndr The N-dimensional rectangle of the current domain
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT capi_return_t tiledb_current_domain_get_ndrectangle(
+    tiledb_current_domain_t* current_domain,
+    tiledb_ndrectangle_t** ndr) TILEDB_NOEXCEPT;
+
+/**
+ * Query if the current domain object is empty or not.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint32_t empty = 0;
+ * tiledb_current_domain_get_is_empty(current_domain, &empty);
+ * @endcode
+ *
+ * @param current_domain The current domain to query
+ * @param is_empty True if nothing is set on the current domain
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT capi_return_t tiledb_current_domain_get_is_empty(
+    tiledb_current_domain_t* current_domain,
+    uint32_t* is_empty) TILEDB_NOEXCEPT;
+
+/**
+ * Query the type of current domain set on the object
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_current_domain_type_t type;
+ * tiledb_current_domain_get_type(current_domain, &type);
+ * @endcode
+ *
+ * @param current_domain The current domain to query
+ * @param type The type of representation set on the current domain
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT capi_return_t tiledb_current_domain_get_type(
+    tiledb_current_domain_t* current_domain,
+    tiledb_current_domain_type_t* type) TILEDB_NOEXCEPT;
```

Comment:
~~Did you forget to add a context parameter?~~ What a delight! We finally start having APIs without a context parameter.